### ISSUE TITLE
Could `piper`'s `Arc` eventually be replaced by implementing `AsyncWrite/Read` for std `Arc` in `futures-io`?

### ISSUE THREAD

**Issue by mitchmindtree** [stjepang/piper#4]
Hi!

I was curious about how piper's `Arc` is different to std's and noticed that it seems to be a simple a wrapper providing the async IO trait implementations. Do you think it would be reasonable to upstream these implementations for the std `Arc` in the `futures-io` crate? Or is this not really feasible for some reason?

PS: thanks for all your work in the async world!

**Comment by stjepang** [comment]
Good point -- I've just submitted this :) https://github.com/rust-lang/futures-rs/pull/2139
